src/test/java/com/adobe/target/edge/client/service/DefaultTargetServiceTest.java

```diff
@@ -209,6 +211,36 @@ public void testExecuteNotificationAsync() {
             any(Map.class), any(String.class), any(DeliveryRequest.class), any(Class.class));
   }
 
+  @Test
+  public void testContextCleanUpBeforeCallToDeliveryApi() {
```

Comment:
Let's add an inverse test case as a sanity check to make sure when the geo values _are_ set they remain as-is after the cleanup method is called.